dart/lib/src/enricher/_web_enricher.dart

```diff
@@ -0,0 +1,149 @@
+    );
+  }
+
+  Future<SentryDevice> _getDevice(SentryDevice? device) async {
+    final batteryManager =
+        await _navigator.getBattery() as html.BatteryManager?;
+    var level = batteryManager?.level;
+    if (level != null) {
+      // batteryManager?.level is a value between 0 and 1
+      level = level * 100;
+    }
+    final charging = batteryManager?.charging;
+
+    // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/deviceMemory
+    final size = _navigator.deviceMemory?.toDouble();
+    final memoryByteSize = size != null ? size * 1024 * 1024 * 1024 : null;
+
+    // https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation
+    SentryOrientation? orientation;
+    final screenOrientation = _screen?.orientation;
+    if (screenOrientation != null) {
+      if (screenOrientation.type?.startsWith('portrait') ?? false) {
+        orientation = SentryOrientation.portrait;
+      }
+      if (screenOrientation.type?.startsWith('landscape') ?? false) {
+        orientation = SentryOrientation.landscape;
+      }
+    }
+
+    String? screenResolution;
+    var screen = _screen;
+    if (screen != null) {
+      screenResolution = '${screen.width}x${screen.height}';
+    }
+
+    /* TODO storage
+    final storage = await _navigator.storage?.estimate();
+    final availableStorage = storage?['quota'];
+    final usedStorage = storage?['usage'];
+    */
+
+    if (device == null) {
+      return SentryDevice(
+        batteryLevel: level?.toDouble(),
+        charging: charging,
+        online: _navigator.onLine,
+        memorySize: memoryByteSize?.toInt(),
+        orientation: orientation,
+        screenResolution: screenResolution,
+        screenDensity: _window.devicePixelRatio.toDouble(),
+      );
+    }
+    return device.copyWith(
+      batteryLevel: level?.toDouble(),
+      charging: charging,
+      online: _navigator.onLine,
+      memorySize: memoryByteSize?.toInt(),
+      orientation: orientation,
+      screenResolution: screenResolution,
+      screenDensity: _window.devicePixelRatio.toDouble(),
+    );
+  }
+
+  SentryBrowser _getBrowser(SentryBrowser? browser) {
+    // TODO: Figure out how to send the correct name and version
+    if (browser == null) {
+      return SentryBrowser(
+        name: _navigator.appName,
+        version: _navigator.userAgent,
+      );
+    } else {
+      return browser.copyWith(
+        name: _navigator.appName,
+        version: _navigator.appVersion,
+      );
+    }
+  }
+
+  SentryOperatingSystem _getOperatingSystem(SentryOperatingSystem? os) {
+    if (os == null) {
+      return SentryOperatingSystem(
+        name: _platform.operatingSystem,
+      );
+    } else {
+      return os.copyWith(
+        name: _platform.operatingSystem,
+      );
+    }
+  }
+
+  List<SentryRuntime> _getRuntimes(List<SentryRuntime>? runtimes) {
+    final dartRuntime = SentryRuntime(name: 'Dart');
+    final flutterRuntime = SentryRuntime(name: 'Flutter');
```

Comment:
This is not Flutter

Reply:
Here we're on dart2js or the other one for dev, right?
Maybe worth adding a key under the Runtime interface like: `Compiler: dart2js` or something?

Reply:
Good point. I believe on web there is only dart2js. On the other platforms there is the VM for debug and AOT for release mode. I'm not sure what's used in profile mode. I'll have to look it up.